### ISSUE TITLE
Ansible treats empty groups as hosts

### ISSUE THREAD

**Issue by RusDavies** [RusDavies/moose-inventory#9]
Create an empty group

``` shell
$ moose-inventory group add empty-group
```

Create the following play book as example.yml:

``` shell

---
- name: example
  hosts: empty-group
  tasks:
     - shell: echo "hello"
```

Now run the playbook:

``` shell
$ ansible-playbook example.yml
```

In the above, I'd expect Ansible to ignore empty groups.  Instead, here's the error we get:

``` shell
PLAY [example] **************************************************************** 

GATHERING FACTS *************************************************************** 
fatal: [empty-group] => SSH Error: data could not be sent to the remote host. Make sure this host can be reached over ssh

TASK: [shell echo "hello"] **************************************************** 
FATAL: no hosts matched or all hosts have already failed -- aborting


PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/home/russell/example.retry

empty-group                : ok=0    changed=0    unreachable=1    failed=0 
```

If we list the hosts Ansible things it has, then we can see something is wrong.

``` shell
$ ansible-playbook example.yml --list-hosts

playbook: example.yml

  play #1 (example): host count=1
    empty-group
```

It seems that Ansible thinks our empty group is a host.  Are we doing something wrong?

**Comment by RusDavies** [comment]
By comparison, if we create a hosts file, and populate it as follows, (and remembering to set our ansible.cfg file to point to hosts instead of inventory),

``` shell
[empty-group]

```

and then have Ansible again list the hosts, then we get the expected result:

``` shell
$ ansible-playbook example.yml --list-hosts

playbook: example.yml

  play #1 (example): host count=0

```

This looks like some oddity about how Ansible creates the structure from out json output.  Perhaps we shouldn't show Ansible any empty groups ?

**Comment by RusDavies** [comment]
... or perhaps we should always show he hosts element, even if empty?  At the moment, we hide it if empty.

**Comment by RusDavies** [comment]
Fixed in 1.0.4.  Simply force output of hosts elements, even if they're empty, whenever the ansible flag is set.
